Sources/Workspace/Workspace.swift

```diff
@@ -1566,26 +1572,79 @@ extension Workspace {
-        var didDownloadAnyArtifact = false
+        // Netrc feature currently only supported on macOS
         #if os(macOS)
-        // Netrc feature currently only supported on macOS 10.13+ due to dependency
-        // on NSTextCheckingResult.range(with:)
-        if #available(macOS 10.13, *) {
-            authProvider = try? Netrc.load(fromFileAtPath: netrcFilePath).get()
-        }
+        authProvider = try? Netrc.load(fromFileAtPath: netrcFilePath).get()
         #endif
 
-        let result = ThreadSafeArrayStore<ManagedArtifact>()
+        // zip files to download
+        // stored in a thread-safe way as we may fetch more from "ari" files
+        let zipArtifacts = ThreadSafeArrayStore<RemoteArtifact>(artifacts.filter { $0.url.pathExtension.lowercased() == "zip" })
+
+        // fetch and parse "ari" files, if any
+        let indexFiles = artifacts.filter { $0.url.pathExtension.lowercased() == "ari" }
+        if !indexFiles.isEmpty {
+            let hostToolchain = try UserToolchain(destination: .hostDestination())
+            let jsonDecoder = JSONDecoder.makeWithDefaults()
+            for indexFile in indexFiles {
+                group.enter()
+                var request = HTTPClient.Request(method: .get, url: indexFile.url)
+                request.options.validResponseCodes = [200]
+                request.options.authorizationProvider = authProvider?.authorization(for:)
+                self.httpClient.execute(request) { result in
+                    defer { group.leave() }
 
-        for artifact in artifacts {
-            group.enter()
-            defer { group.leave() }
+                    do {
+                        switch result {
+                        case .failure(let error):
+                            throw error
+                        case .success(let response):
+                            guard let body = response.body else {
+                                throw StringError("Body is empty")
+                            }
+                            // FIXME: would be nice if checksumAlgorithm.hash took Data directly
+                            let bodyChecksum = self.checksumAlgorithm.hash(ByteString(body)).hexadecimalRepresentation
+                            guard bodyChecksum == indexFile.checksum else {
+                                throw StringError("checksum of downloaded artifact of binary target '\(indexFile.targetName)' (\(bodyChecksum)) does not match checksum specified by the manifest (\(indexFile.checksum ))")
+                            }
+                            let metadata = try jsonDecoder.decode(ArchiveIndexFile.self, from: body)
+                            // FIXME: this filter needs to become more sophisticated
+                            guard let supportedArchive = metadata.archives.first(where: { $0.fileName.lowercased().hasSuffix(".zip") && $0.supportedTriples.contains(hostToolchain.triple) }) else {
```

Comment:
I wonder if we'll eventually need to extend this to support multiple possible host triples (e.g. running x86_64 binaries in Rosetta), but that would be future improvement.  Just something to keep in mind for later.

Reply:
yes this is definitely a detail we will have to enhance over time - hence the FIXME